Task-54770 : Add configuration for SameSite option (#313)
exoplatform/exo-documentation: docs/Security.rst

            ...

.. _Security.SameSiteConfiguration:

======================
SameSite Configuration
======================

SameSite is a property set on HTTP cookies. It can prevent some CSRF attacks.
SameSite property can take one of theses values : None, Strict, and Lax

With value **None**, when a request is done on eXo Server, there is no verification on the referer. The cookies is used.
For example, if a user receive a forged email, containing a link to call a REST allowing to modify data, like delete a space,
if the user have a session on eXo, when he clicks on the link, the session cookie is used and the request is executed. It is a
CSRF attack.

> With value **Strict**; when a request is done on eXo Server, the referer is checked. If the referer have a different

Reviewer: ```suggestion
With value **Strict**; when a request is done on eXo Server, the referer is checked. If the referer has a different
```

domain than the eXo Server domain, the request will not use the cookie.
In the above example, the request is not directly executed, and the user is redirected to the login URL.

> This behaviour is also applied for GET HTTP requests. So, when a user click on a link in a notification for example, he have to

Reviewer: ```suggestion
This behaviour is also applied for HTTP GET requests. So, when a user clicks on a link in a notification for example, he has to
```

> login again.

Author: ```suggestion
login again. 
With this value, all SSO systems (SAML, OAuth, OpenIdConnect ...), generally based on redirections between different hosts, **will not work**.
```

> With value **Lax**; when a request is done on eXo Server, the referer is checked. If the referer have a different

Reviewer: ```suggestion
With value **Lax**; when a request is done on eXo Server, the referer is checked. If the referer has a different
```

domain than the eXo Server domain, and if the request is not a GET request, the request will not use the cookie. So this

> intermediate option allows to use link in mail notification, and protect sensible urls which modify data.

Reviewer: ```suggestion
intermediate option allows to use link in mail notification, and protects sensitive urls which modify data.
```

By default, eXo use **Lax** value for cookies.

> It can be change by configuration if a different value is needed. For that, rename file (if not already done)

Reviewer: ```suggestion
It can be changed by configuration if a different value is needed. For that, rename file (if not already done)
```

            ...

.. _Security.SameSiteConfiguration:

======================
SameSite Configuration
======================

SameSite is a property set on HTTP cookies. It can prevent some CSRF attacks.
SameSite property can take one of theses values : None, Strict, and Lax

With value **None**, when a request is done on eXo Server, there is no verification on the referer. The cookies is used.

> For example, if a user receive a forged email, containing a link to call a REST allowing to modify data, like delete a space,

Reviewer: ```suggestion
For example, if a user receives a malicious email, containing a link forged to call a REST endpoint altering data, like deleting a space, changing a permission, etc..
```

> For example, if a user receives a malicious email, containing a link forged to call a REST endpoint altering data, like deleting a space, changing a permission, etc..

Reviewer: ```suggestion
For example, when a user receives a malicious email, containing a link forged to call a data-altering REST endpoint such as deleting a space, changing a permission, etc..
```

Author: Is there a typo in your suggestion ? the word "dans" should be removed, right ?

```suggestion
For example, when a user receives a malicious email, containing a link forged to call a data-altering REST endpoint altering data, such as deleting a space, changing a permission, etc..
```

Reviewer: yes but tour suggestion still has a repetition for data altering. I updated mine. please review

> if the user have a session on eXo, when he clicks on the link, the session cookie is used and the request is executed. It is a

Reviewer: ```suggestion
if the user has a valid session on eXo, when he clicks on the link, the session cookie would be used and the request would executed on his behalf. It is a
```

> if the user has a valid session on eXo, when he clicks on the link, the session cookie would be used and the request would executed on his behalf. It is a

Reviewer: ```suggestion
if the user has a valid session on eXo, clicking the link would alter data on their behalf. It is a
```

CSRF attack.

> With value **Strict**; when a request is done on eXo Server, the referer is checked. If the referer has a different

Reviewer: ```suggestion
With value **Strict**; when a request arrives on the eXo server, the referer is verified. If the referer has a different
```

> domain than the eXo Server domain, the request will not use the cookie.

Reviewer: ```suggestion
domain than the eXo server's domain, the request will not use the cookie.
```

> In the above example, the request is not directly executed, and the user is redirected to the login URL.

Reviewer: ```suggestion
In the situation described above, the request would not  be directly executed. The user would be redirected to the login page first.
```

This behaviour is also applied for HTTP GET requests. So, when a user clicks on a link in a notification for example, he has to
login again. 
With this value, all SSO systems (SAML, OAuth, OpenIdConnect ...), generally based on redirections between different hosts, **will not work**.

> With value **Lax**; when a request is done on eXo Server, the referer is checked. If the referer has a different

Reviewer: ```suggestion
With value **Lax**; when a request arrives on the eXo server, the referer is also verified. If the referer has a different
```

> domain than the eXo Server domain, and if the request is not a GET request, the request will not use the cookie. So this

Reviewer: ```suggestion
domain than the eXo server's domain, only GET requests will use the cookie. So this
```

> intermediate option allows to use link in mail notification, and protects sensitive urls which modify data.

Reviewer: ```suggestion
intermediate option allows to use links read only endpoints in email notifications, and still protect sensitive requests that may alter data.
```

> By default, eXo use **Lax** value for cookies.

Reviewer: ```suggestion
By default, eXo uses **Lax** policy in order to have email notification links work on SSO contexts.
```

Author: Links in email and SSO contexts are not related. 
With Strict mode, links in notification will request for a new login. And the SSO will not work.
With Lax mode, links in notification will correctly open the page. And SSO will work.

```suggestion
By default, eXo uses **Lax** policy in order to have email notification links and SSO configurations work.
```

WDYT ?
